### PR TITLE
Separate supplemental resources from materials section

### PR DESCRIPTION
**Pull request by vsharma20v**
Separate supplemental resources from the materials section and also fixes some typo errors in the following files:
- [x]  https://github.com/Techtonica/curriculum/blob/master/agile-development/scrum.md
- [x]  https://github.com/Techtonica/curriculum/blob/master/databases/mongo-db.md
- [x]  https://github.com/Techtonica/curriculum/blob/master/security/web-security.md
- [x]  https://github.com/Techtonica/curriculum/blob/master/oauth/o-auth.md
- [x]  https://github.com/Techtonica/curriculum/blob/master/testing-and-tdd/enzyme-testing.md
- [x]  https://github.com/Techtonica/curriculum/blob/master/deploying/deploying.md
- [x]  https://github.com/Techtonica/curriculum/blob/master/recursion/recursion.md
- [x]  https://github.com/Techtonica/curriculum/blob/master/runtime-complexity/memoization.md
- [x]  https://github.com/Techtonica/curriculum/blob/master/agile-development/intro-to-agile.md

I'm unable to find this file
 - [ ]  [https://github.com/Techtonica/curriculum/blob/master/testing-and-tdd/unit-testing.md ](https://github.com/Techtonica/curriculum/blob/master/testing-and-tdd/unit-testing.md)

Issue Fixed: #831

### CODE REVIEW

**Comment by alodahl** [comment]
@vsharma20v good eye!  We moved it to https://github.com/Techtonica/curriculum/blob/master/testing-and-tdd/integration-testing.md recently.

File: testing-and-tdd/enzyme-testing.md [hunk 1]
```diff
@@ -35,23 +38,28 @@ Here are links to lessons that should be completed before this lesson:
...
+- Enzyme & Jest testing series by Wasura Wattearachchi
+    - [Testing with Jest and Enzyme in React-Part 1 (How to configure and run a test using Jest?) (5 mins read)](https://blog.usejournal.com/testing-with-jest-and-enzyme-in-react-part-1-162ce7466128)
+    - [Testing with Jest and Enzyme in React-Part 2 (How to integrate Enzyme with Jest in Testing?) (5 mins read)](https://blog.usejournal.com/testing-with-jest-and-enzyme-in-react-part-2-how-to-integrate-enzyme-with-jest-in-testing-b88917787285)
+    - [Testing with Jest and Enzyme in React-Part 3 (Best Practices when testing with Jest and Enzyme) (6 mins read)](https://blog.usejournal.com/testing-with-jest-and-enzyme-in-react-part-3-best-practices-when-testing-with-jest-and-enzyme-ae3fe0c39d06)
+    - [Testing with Jest and Enzyme in React-Part 4 (shallow vs. mount in Enzyme) (7 mins read)](https://blog.usejournal.com/testing-with-jest-and-enzyme-in-react-part-4-shallow-vs-mount-in-enzyme-d60cad73f85c)
+    - [Testing with Jest and Enzyme in React-Part 5 (Dive with Jest and Enzyme) (7 mins read)](https://blog.usejournal.com/testing-with-jest-and-enzyme-in-react-part-5-dive-with-jest-and-enzyme-85a0d681d173)
```
Comment (alodahl) [inline testing-and-tdd/enzyme-testing.md]: Remove 5 & 6, please - we are only asking apprentices to do parts 1-4.

File: testing-and-tdd/enzyme-testing.md [hunk 2]
```diff
@@ -35,23 +38,28 @@ Here are links to lessons that should be completed before this lesson:
...
+    - [Testing with Jest and Enzyme in React-Part 6 (Snapshot Testing in Jest) (9 mins read)](https://blog.usejournal.com/testing-with-jest-and-enzyme-in-react-part-6-snapshot-testing-in-jest-72fb0ce91c5a)
+- [Enzyme Video by JS Playground (10 mins watch)](https://www.youtube.com/watch?v=nvL2ha0XUYo)
 
 ### Lesson
+- Read through lesson slides [Enzyme tests](https://docs.google.com/presentation/d/1plJI0Sdx4WiXfSWLHiTVyYxvXaMsa0_HVUGtwpgByEE/edit?usp=sharing)
+- Watch  [Enzyme Tutorial - How to Write Test Code for React (10 mins watch)](https://www.youtube.com/watch?v=nvL2ha0XUYo)
+- Work through all parts of the "Testing with Jest & Enzyme in React" series by Wasura Wattearachchi](https://medium.com/@wasuradananjith). This should take about 30 minutes.
 
-Work through parts 1, 2, 3 & 4 of [this "Testing with Jest & Enzyme in React" tutorial by Wasura Wattearachchi](https://medium.com/@wasuradananjith). This should take about 30 minutes.
```
Comment (alodahl) [inline testing-and-tdd/enzyme-testing.md]: Please keep the part about only requiring parts 1, 2, 3 & 4

File: security/web-security.md [hunk 3]
```diff
@@ -28,23 +28,24 @@ Some of the popular companies that use web security are mentioned below with the
...
+    - Cross-site scripting (XSS)
+    - Cross-site request forgery (CSRF)
 
 ### Materials
 
 - [Security Lesson Slideshow](https://docs.google.com/presentation/d/1mNyxzYGW-6M5yxBfJsxiwb9lcko5sa91thlBPKTodMg/edit?usp=sharing)
 - [Techtonica's Roles in Tech Presentation](https://drive.google.com/file/d/1jZY4K-KeqLDM4AXgxwymJrBxGi3DIhaH/view?usp=sharing) (security section starts at 2:57)
-- [7 min read: A quick introduction to web security [FreeCodeCamp]](https://medium.freecodecamp.org/a-quick-introduction-to-web-security-f90beaf4dd41)
 - [Slideshow: Authentication vs Authorization](https://docs.google.com/presentation/d/1iHXimPdzKOYpjhXC2Vh-8QmpG90PxHWBQ-gRb6k32zg/edit?usp=sharing).
+- [7 min read: A quick introduction to web security [FreeCodeCamp]](https://medium.freecodecamp.org/a-quick-introduction-to-web-security-f90beaf4dd41)
```
Comment (saberprashant) [inline security/web-security.md]: 🎯

**Pull request merged**